File: 2022/Day 4/pt1.cpp
#include "helper.hpp"

int get_full_overlaps(std::vector<std::string> lines)
{
    int full_overlaps = 0;

    for (std::string line : lines)
    {
        int l0, l1, r0, r1;
        sscanf(line.c_str(), "%d-%d,%d-%d", &l0, &l1, &r0, &r1);
        full_overlaps += ((l0 <= r0 && l1 >= r1) || (r0 <= l0 && r1 >= l1));
    }

    return full_overlaps;
}

int main()
{
    std::vector<std::string> lines = Helper::get_lines("bigboy.txt");

    auto t1 = Helper::get_current_time();
    int full_overlaps = get_full_overlaps(lines);
    std::cout << "Full Overlaps: " << full_overlaps << std::endl;
    auto t2 = Helper::get_current_time();
    Helper::get_time_taken(t1, t2);

    return 0;
}

File: 2023/Day 01/pt1.go
package main

import (
	"fmt"
	"strconv"

	"github.com/benjababe/advent-of-code/helper"
)

func findFirst(line string) int {
	for len(line) > 0 {
		for num := int64(1); num < 10; num++ {
			if string(line[0]) == strconv.FormatInt(num, 10) {
				return int(num)
			}
		}

		line = line[1:]
	}

	return -1
}

func findLast(line string) int {
	for len(line) > 0 {
		lastI := len(line) - 1

		for num := int64(1); num < 10; num++ {
			if string(line[lastI]) == strconv.FormatInt(num, 10) {
				return int(num)
			}
		}

		line = line[:lastI]
	}

	return -1
}

func solve(lines []string) int {
	score := 0

	for _, line := range lines {
		start, last := 0, 0

		for i := 0; i < len(line); i++ {
			char := line[i]
			if '0' <= char && char <= '9' {
				if start == 0 {
					start = int(char - '0')
				}
				last = int(char - '0')
			}
		}

		score += start*10 + last
	}

	return score
}

func main() {
	lines := []string{}
	helper.GetLines(&lines, "input.txt")

	start := helper.GetCurrentTime()
	output := solve(lines)
	fmt.Printf("Output: %d\n", output)
	end := helper.GetCurrentTime()
	helper.GetTimeTaken(start, end)

	helper.CopyClipboard(strconv.Itoa(output))
}
